sql/core/src/test/resources/sql-tests/inputs/pgSQL/with.sql

```diff
@@ -0,0 +1,1222 @@
+--(t1.id=t2.id);
+
+--
+-- test cycle detection
+--
+-- [ORIGINAL SQL]
+--create temp table graph( f int, t int, label text );
+create table graph( f int, t int, label string ) USING parquet;
+
+insert into graph values
+	(1, 2, 'arc 1 -> 2'),
+	(1, 3, 'arc 1 -> 3'),
+	(2, 3, 'arc 2 -> 3'),
+	(1, 4, 'arc 1 -> 4'),
+	(4, 5, 'arc 4 -> 5'),
+	(5, 1, 'arc 5 -> 1');
+
+-- [SPARK-24497] Support recursive SQL query
+--with recursive search_graph(f, t, label, path, cycle) as (
+--	select *, array[row(g.f, g.t)], false from graph g
+--	union all
+--	select g.*, path || row(g.f, g.t), row(g.f, g.t) = any(path)
+--	from graph g, search_graph sg
+--	where g.f = sg.t and not cycle
+--)
+--select * from search_graph;
+
+-- ordering by the path column has same effect as SEARCH DEPTH FIRST
+-- [SPARK-24497] Support recursive SQL query
+--with recursive search_graph(f, t, label, path, cycle) as (
+--	select *, array[row(g.f, g.t)], false from graph g
+--	union all
+--	select g.*, path || row(g.f, g.t), row(g.f, g.t) = any(path)
+--	from graph g, search_graph sg
+--	where g.f = sg.t and not cycle
+--)
+--select * from search_graph order by path;
+
+--
+-- test multiple WITH queries
+--
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--  y (id) AS (VALUES (1)),
+--  x (id) AS (SELECT * FROM y UNION ALL SELECT id+1 FROM x WHERE id < 5)
+--SELECT * FROM x;
+
+-- forward reference OK
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--    x(id) AS (SELECT * FROM y UNION ALL SELECT id+1 FROM x WHERE id < 5),
+--    y(id) AS (values (1))
+-- SELECT * FROM x;
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--   x(id) AS
+--     (VALUES (1) UNION ALL SELECT id+1 FROM x WHERE id < 5),
+--   y(id) AS
+--     (VALUES (1) UNION ALL SELECT id+1 FROM y WHERE id < 10)
+-- SELECT y.*, x.* FROM y LEFT JOIN x USING (id);
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--   x(id) AS
+--     (VALUES (1) UNION ALL SELECT id+1 FROM x WHERE id < 5),
+--   y(id) AS
+--     (VALUES (1) UNION ALL SELECT id+1 FROM x WHERE id < 10)
+-- SELECT y.*, x.* FROM y LEFT JOIN x USING (id);
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--   x(id) AS
+--     (SELECT 1 UNION ALL SELECT id+1 FROM x WHERE id < 3 ),
+--   y(id) AS
+--     (SELECT * FROM x UNION ALL SELECT * FROM x),
+--   z(id) AS
+--     (SELECT * FROM x UNION ALL SELECT id+1 FROM z WHERE id < 10)
+-- SELECT * FROM z;
+
+-- [SPARK-24497] Support recursive SQL query
+--WITH RECURSIVE
+--   x(id) AS
+--     (SELECT 1 UNION ALL SELECT id+1 FROM x WHERE id < 3 ),
+--   y(id) AS
+--     (SELECT * FROM x UNION ALL SELECT * FROM x),
+--   z(id) AS
+--     (SELECT * FROM y UNION ALL SELECT id+1 FROM z WHERE id < 10)
+-- SELECT * FROM z;
+
+--
+-- Test WITH attached to a data-modifying statement
+--
+
+-- [ORIGINAL SQL]
+--CREATE TEMPORARY TABLE y (a INTEGER) USING parquet;
+CREATE TABLE y (a INTEGER) USING parquet;
```

Comment:
BTW, please keep `TEMPORARY` like the original SQL statement. Then, you can remove `DROP TABLE` at line 477.
```
spark-sql> CREATE TEMPORARY TABLE y (a INTEGER) USING parquet;
Time taken: 1.144 seconds
```

Reply:
I will recheck all `CREATE TABLE`s but the reason I dropped `TEMPORARY` is that `INSERT INTO` follows most of these tables so they can't be temporary.

Reply:
Done.